src/diart/models.py

```diff
@@ -143,33 +162,25 @@ def from_pyannote(
         assert _has_pyannote, "No pyannote.audio installation found"
         return PyannoteEmbeddingModel(model, use_hf_token)
 
-    @abstractmethod
-    def forward(
-        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
-    ) -> torch.Tensor:
-        """
-        Forward pass of an embedding model with optional weights.
-
-        Parameters
-        ----------
-        waveform: torch.Tensor, shape (batch, channels, samples)
-        weights: Optional[torch.Tensor], shape (batch, frames)
-            Temporal weights for each sample in the batch. Defaults to no weights.
-
-        Returns
-        -------
-        speaker_embeddings: torch.Tensor, shape (batch, embedding_dim)
-        """
-        pass
-
 
 class PyannoteEmbeddingModel(EmbeddingModel):
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__(PyannoteLoader(model_info, hf_token))
 
-    def forward(
-        self,
-        waveform: torch.Tensor,
-        weights: Optional[torch.Tensor] = None,
+    def __call__(
+        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        return self.model(waveform, weights=weights)
+        if not isinstance(self.model, WeSpeakerPretrainedSpeakerEmbedding):
+            return super().__call__(waveform, weights)
+        else:
+            self.load()
+            # Normalize weights
+            if weights is not None:
+                weights -= weights.min(dim=1, keepdim=True).values
+                weights /= weights.max(dim=1, keepdim=True).values
```

Comment:
Same comment as before, you need to divide by `max - min`, and not by `max`

Reply:
Using `weights -= weights.min(dim=1, keepdim=True).values` the line before should ensure that `min()` is equal to 0.
Nonetheless, I can add `-min` if you want.

Reply:
See [the normalization equation](https://en.m.wikipedia.org/wiki/Feature_scaling) under section Methods, Rescaling (min max normalization).

I think you're missing the "- min" from the denominator

Reply:
Closing this since you've fixed it.